### PR TITLE
Sort and merge incoming telemetry

### LINKED ISSUE #5102: Multiple telemetries destroy table performance
#### Summary
For tables with multiple telemetries added, combining data into the table is not performant. The datasets are combined individually, and each additional dataset takes exponentially longer than the last.

#### Expected vs Current Behavior
Tables should load and render rows in a reasonable amount of time

#### Steps to Reproduce

1. Create a telemetry table and add several telemetry endpoints (5 to 10)
2. Query a time period (start small) using the time conductor (fixed time is easier to observe and control amount of datapoints)
3. Observe time it takes for each dataset to render, shown by the rows count in the lower right of the table updating
4. Change time window to increase amount of datapoints if time to render each set of data is negligible
5. Can also compare to an overlay plot with same data for kicks

#### Environment

* Open MCT Version: 
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?

### PR DESCRIPTION
**Pull request by davetsay**
Closes #5102
Filed #5101 for additional tests to test sort order

### Describe your changes:
* use sort and merge for incoming telemetry

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/plugins/telemetryTable/collections/TableRowCollection.js [hunk 1]
```diff
@@ -92,59 +89,86 @@ define(
...
+            sortAndMergeRows(rows) {
+                const sortedRowsToAdd = this.sortCollection(rows);
 
-            /**
-             * Finds the correct insertion point for the given row.
-             * Leverages lodash's `sortedIndex` function which implements a binary search.
-             * @private
-             */
-            sortedIndex(rows, testRow, lodashFunction = _.sortedIndexBy) {
                 if (this.rows.length === 0) {
```
Comment (akhenry) [inline src/plugins/telemetryTable/collections/TableRowCollection.js:95]: This feels like an unexpected side-effect of the function, and I know it already existed in the code. In the interests of time let's file a followup to fix it. The calling code should be explicitly resetting the TableRowCollection.

File: src/plugins/telemetryTable/collections/TableRowCollection.js [hunk 2]
```diff
@@ -92,59 +89,86 @@ define(
...
+                return sortedRows;
+            }
+
+            mergeSortedRows(rows) {
+                const mergedRows = [];
+                let i = 0;
+                let j = 0;
+
+                while (i < this.rows.length && j < rows.length) {
+                    const iRow = this.rows[i];
```
Comment (akhenry) [inline src/plugins/telemetryTable/collections/TableRowCollection.js:134]: `existingRow` maybe?

File: src/plugins/telemetryTable/collections/TableRowCollection.js [hunk 3]
```diff
@@ -92,59 +89,86 @@ define(
...
+                    const jRow = rows[j];
```
Comment (akhenry) [inline src/plugins/telemetryTable/collections/TableRowCollection.js:135]: `incomingRow` maybe?

**Review by akhenry — APPROVED** [review]

File: src/plugins/telemetryTable/collections/TableRowCollection.js (hunk 2 above)
Comment (davetsay) [reply src/plugins/telemetryTable/collections/TableRowCollection.js:134]: done much better

File: src/plugins/telemetryTable/collections/TableRowCollection.js (hunk 3 above)
Comment (davetsay) [reply src/plugins/telemetryTable/collections/TableRowCollection.js:135]: done much better

**Pull request merged**